Assets/MRTK/SDK/Experimental/Joystick/JoystickController.cs

```diff
@@ -0,0 +1,153 @@
+
+        Vector3 startPosition;
+        Vector3 joystickGrabberPosition;
+        Vector3 joystickVisualRotation;
+        const int joystickVisualMaxRotation = 80;
+        bool isDragging = false;
+
+
+        private void Start()
+        {
+            startPosition = GrabberVisual.transform.position;
+            if(GrabberVisual != null)
+            {
+                GrabberVisual.GetComponent<MeshRenderer>().enabled = ShowGrabberVisual;
+            }
+        }
+        void Update()
+        {
+            if (!isDragging)
+            {
+                // when dragging stops, move joystick back to idle
+                if(GrabberVisual != null)
+                {
+                    GrabberVisual.transform.position = Vector3.Lerp(GrabberVisual.transform.position, startPosition, Time.deltaTime * ReboundSpeed);
+                }
+            }
+            calculateJoystickRotation();
+            applyJoystickValues();
+        }
+        void calculateJoystickRotation()
+        {
+            joystickGrabberPosition = GrabberVisual.transform.position - startPosition;
+            // Left Right
+            joystickVisualRotation.z = Mathf.Clamp(-joystickGrabberPosition.x * SensitivityLeftRight,-joystickVisualMaxRotation, joystickVisualMaxRotation);
+            // Forward Back
+            joystickVisualRotation.x = Mathf.Clamp(joystickGrabberPosition.z * SensitivityForwardBack,-joystickVisualMaxRotation, joystickVisualMaxRotation);
+            if (JoystickVisual != null)
+            {
+                JoystickVisual.transform.localRotation = Quaternion.Euler(joystickVisualRotation);
+            }
+        }
+        void applyJoystickValues()
+        {
+            if(ObjectToManipulate != null)
+            {
+                if (mode == JoystickMode.Move)
+                {
+                    ObjectToManipulate.transform.position += (joystickGrabberPosition * MultiplierMove);
+                    if (DebugText != null)
+                    {
+                        DebugText.text = ObjectToManipulate.transform.position.ToString();
+                    }
+                }
+                else if (mode == JoystickMode.Rotate)
+                {
+                    Vector3 newRotation = ObjectToManipulate.transform.rotation.eulerAngles;
+                    // only take the horizontal axis from the joystick
+                    newRotation.y += (joystickGrabberPosition.x * MultiplierRotate);
+                    newRotation.x = 0;
+                    newRotation.z = 0;
+                    ObjectToManipulate.transform.rotation = Quaternion.Euler(newRotation);
+                    if (DebugText != null)
+                    {
+                        DebugText.text = ObjectToManipulate.transform.rotation.eulerAngles.ToString();
+                    }
+                }
+                else if (mode == JoystickMode.Scale)
+                {
+                    //TODO: Clamp above zero
```

Comment:
The build is failing here. We need a space after all // comments :T
```suggestion
                    // TODO: Clamp above zero
```